### ISSUE TITLE
Difficult to maintain and update a whitelist

### ISSUE THREAD

**Issue by monk-time** [mrdokenny/Cookie-AutoDelete#225]
This is a suggestion about improving the UI/UX of adding new entries to a white/greylist, based mainly on an ancient and dead Firefox Extension [selectivecookiedelete](https://addons.mozilla.org/en-US/firefox/addon/selectivecookiedelete/). It had an ingenious and simple dialog for managing a whitelist (WL) that had several major advantages over CAD's (or any other SDC-like add-on for that matter):

![Image](https://addons.cdn.mozilla.net/user-media/previews/full/30/30896.png?modified=1331218902)

1. No need to type in domains manually, easy to just click/tap on the ones you need to whitelist.
2. The list of domains of existing cookies (not covered by the WL) is visible, showing a preview of which cookies will be removed.
3. Clear separation of domains covered by the WL from those that have left their cookies since the last cleaning. 

The last one is the biggest one because it allows maintaining a WL over time. **If I clean cookies manually say once a month, I may not remember which sites I need to add to the WL by the time I clean them.** But with that UI the panel on the left instantly shows me all recently visited non-WL sites, some of which I can whitelist right there.

A panel of non-WL domains that have left cookies can be displayed on the "List of Expressions" tab below the WL, sorted by second-level domains (so that "google.com" is followed by "maps.google.com" and "www.google.com"). Each row should have buttons or a drop-down menu with actions for adding a full or a partial pattern to the white/greylist, e.g. for "maps.google.com" at least three actions are required: "add maps.google.com to WL", "add google.com to WL", "add *.google.com to WL". The panel should auto-update after clicking any of those actions, removing the rows that are covered by the new WL entry.

**Comment by ke-d** [comment]
Something similar has been proposed which works well with the current UX in #120

**Comment by monk-time** [comment]
@mrdokenny I am sorry, but I don't see anything in that ticket that makes it easier to check which of the new sites visited since the last clean-up I need to add to the whitelist. Autocompleting/filtering the whitelist doesn't help at all since I don't even know which domains to look for.

It is up to you, of course, but I disagree with closing this ticket. I understand I wrote a wall of text in the OP, but I think I've described the use case (quite common, I imagine) fairly well there.
